fastparse/shared/src/main/scala/fastparse/sequence/SeqApi.scala

```diff
@@ -0,0 +1,62 @@
+package fastparse.sequence
+
+import scala.reflect.ClassTag
+import fastparse.core
+import fastparse.parsers
+
+/**
+ * This is a trait applicable on potentially any sequences but Vector or Array
+ * are typically usable in real world.
+ */
+abstract class SeqApi[Elem, Repr <: Seq[Elem]](ct: ClassTag[Elem], reprOps: fastparse.utils.ReprOps[Elem, Repr]) {
```

Comment:
Yes for IndexedSeq[Elem].
On scala.List, it needs to create tests to see if it's really slow.
I think the usage case is when we've a large flat sequence of "primitives" to reorder and reduce, so in my head, access to List is slower than Vector (but maybe I'm wrong) ?